### PR TITLE
kie-issues#1173: Retrieve workflow definitions from data index instead of openapi.json

### LINKED ISSUE #1173: Retrieve workflow definitions from data index instead of openapi.json

### PR DESCRIPTION
**Pull request by paulovmr**
Closes: https://github.com/apache/incubator-kie-issues/issues/1173

### CODE REVIEW

File: packages/runtime-tools-swf-gateway-api/src/gatewayApi/apis.tsx [hunk 1]
```diff
@@ -518,42 +518,28 @@ export const triggerCloudEvent = (event: CloudEventRequest, baseUrl: string, pro
...
+        errorPolicy: "all",
       })
-      .catch((err: any) => reject(err));
+      .then((value) => {
+        const workflowDefinitions = value.data.ProcessDefinitions;
+        resolve(
+          value.data.ProcessDefinitions.map((item: { id: string; endpoint: string }) => {
+            return {
+              workflowName: item.id,
+              endpoint: item.endpoint,
```
Comment (fantonangeli) [inline packages/runtime-tools-swf-gateway-api/src/gatewayApi/apis.tsx:539]: I think we should continue composing the `endpoint` using the `kogitoServiceUrl` to send the correct `Target-Url` to cors-proxy

File: packages/serverless-logic-web-tools/src/runtimeTools/contexts/WebToolsWorkflowDefinitionListContextProvider.tsx [hunk 2]
```diff
@@ -20,12 +20,17 @@
...
 import { useSettings } from "../../settings/SettingsContext";
+import { useEnv } from "../../env/EnvContext";
 
 export function WebToolsWorkflowDefinitionListContextProvider(props: React.PropsWithChildren<{}>) {
   const settings = useSettings();
+  const { env } = useEnv();
 
   return (
-    <WorkflowDefinitionListContextProvider kogitoServiceUrl={settings.runtimeTools.config.kogitoServiceUrl}>
+    <WorkflowDefinitionListContextProvider
```
Comment (fantonangeli) [inline packages/serverless-logic-web-tools/src/runtimeTools/contexts/WebToolsWorkflowDefinitionListContextProvider.tsx:30]: Following my previous comment about `kogitoServiceUrl` I think we still need to pass it to `WorkflowDefinitionListContextProvider`

**Review by fantonangeli — CHANGES_REQUESTED** [review]
@paulovmr Thank you for your PR.
I left 2 comments about `kogitoServiceUrl` which should be needed to send the data to the workflow, but overall looks good.

**Comment by caponetto** [comment]
@paulovmr do we still need to set up this field? 
All the info you need is present on the Data Index, including the Service URLs.

image

File: packages/runtime-tools-swf-gateway-api/src/gatewayApi/apis.tsx (hunk 1 above)
Comment (paulovmr) [reply packages/runtime-tools-swf-gateway-api/src/gatewayApi/apis.tsx:539]: Hi @fantonangeli , if you have `kogito.service.url` set in the application.properties, the endpoint recorded in the data-index will come correctly with the base URL. Like this:
```
{
  "data": {
    "ProcessDefinitions": [
      {
        "id": "expression",
        "endpoint": "http://localhost:8480/expression"
      }
    ]
  }
}
```

**Comment by caponetto** [comment]
@fantonangeli I think we'll need a similar improvement to the SonataFlow web app. WDYT?

**Comment by fantonangeli** [comment]
> @fantonangeli I think we'll need a similar improvement to the SonataFlow web app. WDYT?

Yes, I agree

File: packages/runtime-tools-swf-gateway-api/src/gatewayApi/apis.tsx (hunk 1 above)
Comment (fantonangeli) [reply packages/runtime-tools-swf-gateway-api/src/gatewayApi/apis.tsx:539]: I confirm it works. Thanks for your instruction.

**Comment by pefernan** [comment]
hi @paulovmr ! Would it be possible to bring this to the `jbpm-quarkus-devui` too??

**Comment by paulovmr** [comment]
hi @pefernan ! Sure, I'm doing some other changes in the web tools UI to wrap up this PR, but after I can send one with similar changes to the current ones for jbpm as well.

File: packages/serverless-logic-web-tools/src/navigation/Routes.ts [hunk 3]
```diff
@@ -148,14 +148,20 @@ export const routes = {
...
-  }>(({ workflowId }) => `/runtime-tools/workflow-details/${workflowId}`),
+  }>(({ workflowId }) => `/runtime-tools/workflow-instances/${workflowId}`),
   runtimeToolsWorkflowForm: new Route<{
     pathParams: PathParams.WORKFLOW_NAME;
   }>(({ workflowName }) => `/runtime-tools/workflow-definition/${workflowName}`),
-  runtimeToolsTriggerCloudEvent: new Route<{}>(() => `/runtime-tools/trigger-cloud-event`),
-  runtimeToolsTriggerCloudEventForWorkflow: new Route<{
+  /*  runtimeToolsTriggerCloudEvent: new Route<{
+    queryParams: QueryParams.URL;
+  }>(() => `/runtime-tools/trigger-cloud-event`),*/
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/navigation/Routes.ts]:
```suggestion
```

**Review by caponetto — COMMENTED** [review]
LGTM. Left only one minor comment.

**Comment by fantonangeli** [comment]
LGTM and tested locally. 
Only one question: do we want to update SonataFlow Deployment Webapp in this PR or in a separate PR?
https://github.com/apache/incubator-kie-tools/blob/e35af74d9bda0b444e712294d29363426330e1ff/packages/sonataflow-deployment-webapp/src/pages/Workflows/Workflows.tsx#L43

**Review by caponetto — APPROVED** [review]

**Comment by paulovmr** [comment]
@fantonangeli Thanks, I've created https://github.com/apache/incubator-kie-issues/issues/1276 to tackle the `sonataflow-deployment-webapp` separately.

**Review by fantonangeli — APPROVED** [review]

**Pull request merged**